modules/rostests/apitests/shell32/CShellDesktop.cpp

```diff
@@ -108,22 +110,22 @@ TestCompareIDList(IShellFolder* psf)
 
     compare(psf, dir3, desktop, S_LESSTHAN);
     compare(psf, desktop, dir3, S_GREATERTHAN);
-    compare(psf, dir3, programs, S_GREATERTHAN);
-    compare(psf, programs, dir3, S_LESSTHAN);
+    compare(psf, dir3, programs, g_bVista ? S_LESSTHAN : S_GREATERTHAN);
```

Comment:
If only vista has this behavior, it might just be a setting that is different in your vm?